### ISSUE TITLE
Editor very slow on Mac Yosemite and El Capitan

### ISSUE THREAD

**Issue by Rojj** [spyder-ide/spyder#2724]
## Description

**What steps will reproduce the problem?**
When I try to edit any python file there is a noticeable delay for every keystroke. It's almost unusable. 
The same issue happens with the Anaconda Spyder and the executable downloaded from GitHub.
1. 
2. 
3. 

**What is the expected output? What do you see instead?**

**Please provide any additional information below**

## Version and main components

* Spyder Version: 2.3.6 
* Python Version: 2.7.10
* Qt Versions:  4.8.6, PyQt4 (API v2) 4.11.3 on Darwin

## Optional dependencies
```
pyflakes >=0.5.0:  0.9.2 (OK)
pep8 >=0.6      :  1.6.2 (OK)
pygments >=1.6  :  2.0.2 (OK)
IPython >=1.0   :  4.0.0 (OK)
zmq >=2.1.11    :  14.7.0 (OK)
qtconsole >=4.0 :  4.0.1 (OK)
pandas >=0.13.1 :  0.16.2 (OK)
sphinx >=0.6.6  :  1.3.1 (OK)
psutil >=0.3    :  2.2.1 (OK)
rope >=0.9.2    :  0.9.4 (OK)
jedi >=0.8.1    :  0.9.0 (OK)
matplotlib >=1.0:  1.4.3 (OK)
sympy >=0.7.3   :  0.7.6 (OK)
pylint >=0.25   :  None (NOK)

```

**Comment by ccordoba12** [comment]
This is probably an issue with Qt, which is about to be updated in Anaconda. I'll let you know when that's the case.

**Comment by demcanulty** [comment]
I'm also eagerly awaiting this update.  I love Spyder, it's really radically improved my python programming experience and makes me happier and more productive, but the slowdowns are frustrating.  Thanks for all your work on this, Carlos!

**Comment by ccordoba12** [comment]
Qt was updated in Anaconda a couple of weeks ago. Please let us know if that fixes this problem for you :-)

**Comment by nbara** [comment]
Unfortunately not. I was hoping it would as well.

I use the latest anaconda (2.4.0), on El Capitan (10.1.1).

Qt version (4.8.7)

**Comment by Rojj** [comment]
Same problem here. Using latest anaconda and Yosemite

**Comment by chrisjcameron** [comment]
As an experiment, try disabling all code introspection features in Preferences > Editor : Code Introspection / Analysis. I turned these off due to an issue #1659 with Jedi and Spyder behaves much better overall.

**Comment by Rojj** [comment]
Just tried this. No changes for me.

**Comment by jlmeo** [comment]
I just installed anaconda yesterday and am experiencing the same problem.

**Comment by icam0** [comment]
Me as well

**Comment by pmjandreen** [comment]
I also experience these problems. Spyder is so slow that it is hardly usable at the moment.

**Comment by cchenship** [comment]
same here

**Comment by pmjandreen** [comment]
Here too

**Comment by ProtossProbe** [comment]
Shae here

**Comment by jandyman** [comment]
Same here. Just checking out Spyder - new to Python. The editor is so slow it is unusable. I'll have to look for another environment and hope to come back and try Spyder again at a later date. Macbook Pro Mid 2014, El Capitan

**Comment by chrisjcameron** [comment]
Spyder runs ok for me on a 2010 MBP with Yosemite (OS X 10.10), so this problem is not universal. I imagine it would be helpful if we were able to provide more information about how it is slow and what features, dependencies or hardware configurations are associated with slow performance. 

To check:

 1. Check the optional dependences report in the help menu and see if you are missing any. 
 2. Systematically turn off the editor features in the Preferences.

To report:

 3. Are you having the exact same slowness issue reported by the original OP?
 3. Is the slowness in the editor, console or inspector? 
 4. Is it always slow or does it get slower with document length, time, etc?
 6. Does it affect pasted and typed lines the same way? (Do pasted lines appear immediately)?
 5. Do you still get a slowdown with *All* editor features disabled? (e.g. all Code introspection and Advanced settings disabled)
 8. Does the problem appear in a new file with no modules imported? 



Spyder installed via the standard Anaconda distribution is working for me with the following installed:  
## Version and main components

* Spyder Version: 2.3.8 
* Python Version: 2.7.11
* Qt Versions:  4.8.7, PyQt4 (API v2) 4.11.4 on Darwin

## Optional dependencies
```
pyflakes >=0.5.0   :  1.0.0 (OK)
pep8 >=0.6         :  1.6.2 (OK)
pygments >=1.6     :  2.0.2 (OK)
IPython >=1.0      :  4.0.1 (OK)
zmq >=2.1.11       :  15.1.0 (OK)
qtconsole >=4.0    :  4.1.1 (OK)
pandas >=0.13.1    :  0.17.1 (OK)
sphinx >=0.6.6     :  1.3.1 (OK)
psutil >=0.3       :  3.3.0 (OK)
rope >=0.9.2       :  0.9.4 (OK)
jedi >=0.8.1;<0.9.0:  0.9.0 (NOK)
matplotlib >=1.0   :  1.5.0 (OK)
sympy >=0.7.3      :  0.7.6.1 (OK)
pylint >=0.25      :  1.4.2 (OK)
```

**Comment by ccordoba12** [comment]
@chrisjcameron is right, on my tests I haven't found the kind of slowness you're describing (else I would've solved it already :-)

Another important thing to let us know is if you're using Anaconda, our DMG installer or another method of installation.

**Comment by pmjandreen** [comment]
I normally use the Anaconda Spyder which is the one that is extremely laggy for me at the moment. I tried the DMG installer today and didn't experience the slowness.

**Comment by chrisjcameron** [comment]
@pmjandreen When you use the help and open a bug report, you get a list of installed packages. Are there any differences in version number or packages installed on these reports when you compare the output for the Anaconda and Spyder versions?

Can you verify that the editor settings are exactly the same between the two versions and test each version with only one new document opened?

**Comment by chrisjcameron** [comment]
A comment on #2817 suggests the slowness may be related to using an external monitor. Is this case it seems like people who are not using an external monitor might experience slowness in the editor. For those of you with the issue, are you able to test with an external monitor? If you are on a dual GPU MacBook Pro, can you note which GPU is active when you have the issue?

**Comment by chrisjcameron** [comment]
Update: Typing long gibberish lines at high speeds results in a backlog up to approximately a dozen characters in the Editor. These unwritten characters are written (one at a time) after I stop typing. There is no noticeable delay when pasting the same lines into the editor. Rapid typing in the Spyder iPython console does not produce a noticeable backlog of unwritten characters.  

This behavior seemed to occur with all screen/GPU combinations on my 2010 MBP. Disabling Editor features did not have an impact on the character backlog. 

It seems possible that a very fast typist would notice the delays I am observing. Is there some code specific to the Editor that preprocesses text (or keypress) input as it is entered?

**Comment by jandyman** [comment]
I experience the problem on an external monitor and without. I experience it using the Anaconda launcher, and I just tried installing from DMG and see the same problem. I do not consider myself an _extremely_ fast typist, though I am a touch typist and a have been programming for decades. To me the lag is VERY noticeable - unusable to me. 

When I was testing tonight I could have sworn that I did not see the delay at first, then it started and wouldn't leave. The only thing I can think of that might have changed is the working directory. I don't know the default directory so I couldn't change back for test purposes. 

I only have one file open. There is no delay in the shell window, just the editor window.

There seems to be some linkage between the Anaconda and DMG installs because when I bring up the DMG version, it knows my file history from the Anaconda version. Just FYI.

I'm pretty new to Python so I'll need help if I am to be guinea pig (which I'm very willing to be, I just need specific instructions). 

I don't experience this problem when using IEP. But Spyder looks much more up to date WRT to component versions and doesn't exhibit a couple problems I've seen in IEP, so I'm anxious to try it.
But with the editor delay is it a non starter.

- Andy

**Comment by pmjandreen** [comment]
I experience similarly to @jandyman that the problem shows up gradually and becomes worse and worse. That's why I might not have experienced it with the dmg version. I only used it for a short while to test (and might have come to the wrong conclusion). My colleague sitting next to me experience the same lagging for Spyder on mac by the way (using Anaconda).

@chrisjcameron these are the differences between my Anaconda Spyder and the dmg Spyder:

Anaconda Spyder, dmg Spyder
Python Version:  2.7.11,  Python Version: 2.7.10
zmq: 14.7.0, zmq:  15.1.0
rope: 0.9.4, rope: 0.10.2
jedi: 0.9.0, None
pylint: None, 1.4.4

The editor settings are identical for the two versions.

**Comment by pete86** [comment]
I experience same issue from Spyder via anaconda in El Capitan.

**Comment by chrisjcameron** [comment]
@ccordoba12: Are you able to replicate the experience of a small delay for rapidly typed characters in the editor? When I type characters very quickly I can build up a queue of a dozen or so characters that appear one at a time after I stop typing. The delay is brief but noticeable. In the iPython console, the same rapid typing produces the characters immediately and with no discernible backlog. 

Do you have a sense of how these two UI fields handle input text differently? It may be something that scales with the number of characters or lines in the file.

**Comment by ccordoba12** [comment]
> @ccordoba12: Are you able to replicate the experience of a small delay for rapidly typed characters in the editor?

Yep I was able to do it, but only using the Anaconda packages, not when using our DMG. That implies that this is a problem with Qt/PyQt and not with Spyder itself (given that Qt versions for both packages are different).

However, the delay shows up only if you type *very* fast. I didn't notice any other noticeable delay, as others in this thread mention.

My guess is that the problem depends on Qt and a weird interaction with the operating system or the graphics card. So this seems to be out of our hands, sorry.

**Comment by massimocaccia** [comment]
So Spyder will always be this way from now on?
Its basically impossible to code...

**Comment by pete86** [comment]
I tried the anaconda version and the dmg versions (both beta and standard) and after a few minutes it becomes so slow it's basically unusable. I have tried using the PyCharm IDE instead, I do not encounter any slowdown issues here.

**Comment by ccordoba12** [comment]
@massimocaccia , sorry but I'm not being able to reproduce this problem. Without that, how could I fix it? Besides it seems to affect few people (if this were a widespread problem I'd be hearing a lot more about it :-).

@pete86, Pycharm uses a different graphical toolkit so I'm not surprised is working just fine.

----

To all affected people: please help us to solve this issue by debugging it. One option could be to install Homebrew or Macports, install or compile Qt with any of them, and see if that solves the problem. If not, you could change compiling options and try again.

Given that none of core developers is able to reproduce this bug, reassuring us that this is problem for you is of no value in this case.

**Comment by kcarnold** [comment]
I have two nearly identical systems, both Anaconda py35, editing some of the same files. One slows down several tenths of a second per keystroke after being open for a while (usually after I've run some code, but haven't confirmed that 100%). The other is just fine. The main difference between the two systems is that the laggy one is on El Capitan, and the fine one is Yosemite. (Also the El Capitan machine has some homebrew packages and the Yosemite one has some MacPorts, but that shouldn't matter because I'm using Anaconda for everything Python on both.)

**Comment by chrisjcameron** [comment]
@kcarnold Can you report the computer make/models and graphics cards? You might be in a position to narrow down the hardware configurations that create an issue.

**Comment by massimocaccia** [comment]
Come to think of it, my Spyder is slow since I upgraded to El Capitan

**Comment by kcarnold** [comment]
The working one is older: early-2011 13" with Intel HD Graphics 3000 512 MB. The laggy one is newer: mid-2014 Retina, NVIDIA GeForce GT 750M 2048 MB.

**Comment by chrisjcameron** [comment]
@kcarnold - that is interesting. Can you force the mid-2014 Retina to use the integrated graphics (in system preferences or with a tool like gfxCardStatus and see if it still works?

**Comment by kcarnold** [comment]
I had a spyder session open, and it was being laggy. gfxCardStatus reported discrete GPU. Without closing the session, switched it to integrated graphics. Still laggy.

**Comment by sho-87** [comment]
I am experiencing the same problem. Spyder is impossible to use on my macbook

I picked up a 13" macbook pro a few weeks ago so its running el capitan. I installed anaconda distribution  and spyder is impossibly slow. I thought maybe it was something to do with the version being packaged with anaconda so I uninstalled that and installed straight from the dmg but still experience the same problem others have described here

This is not an issue on any of my windows machines

**Comment by jeizenga** [comment]
Likewise. My machine is a three-week-old MacBook Pro running El Capitan.

**Comment by sho-87** [comment]
Just tested this on a late 2015 iMac (el capitan) in the lab through the anaconda distribution. No slowness problems on that machine. Only seems to happen on my 2015 macbook pro...

**Comment by kcarnold** [comment]
@sho-87 for me it only happens after at least a few minutes of use, possibly including running files in the console. When I first bring up the editor it's fine. So (1) is your MBP also quick at first like mine, and (2) Did you use it long enough on the iMac to tell?

**Comment by sho-87** [comment]
@kcarnold On my macbook pro the problem occurs as soon as I open spyder, even if its the only thing running, after a reboot etc. 

On the iMac there was 1) no problem straight away, and 2) even after waiting/using it for about 10-15mins there was still no problem

Some specs that may help:

13" MBP Retina 2015:
- El capitan
- 2.9Ghz i5
- 16gb RAM
- Iris 6100
- python 2.7.10
- spyder 2.3.8 (through dmg, not anaconda - although the same problem occurs with the anaconda distribution as I mentioned)

21.5" iMac 2015:
- El capitan
- 2.8Ghz i5
- 8gb RAM
- Iris Pro 6200
- python 2.7.11 (anaconda)
- spyder 2.3.8 (anaconda)

**Comment by psemyon** [comment]
I experience the same issue here. 
Macbook pro 2015 retina

- El Capitan
- 2,2 GHz Intel Core i7
- Intel Iris Pro 1536 MB

I have detected a very clear dependency, when this bug starts accuring in my case:
I open freshly new spyder window, create new file, and just after i type any method (e.g. np.random() ) from imported library, spyder outputs suggestion bubble, and after that the editor becomes as slow, as described above. When i tried this:

> As an experiment, try disabling all code introspection features in Preferences > Editor : Code Introspection / Analysis. I turned these off due to an issue #1659 with Jedi and Spyder behaves much better overall.

it helped.

Dependencies are following:
IPython >=1.0      :  4.0.3 (OK)
jedi >=0.8.1;<0.9.0:  0.9.0 (NOK)
matplotlib >=1.0   :  1.5.1 (OK)
pandas >=0.13.1    :  0.17.1 (OK)
pep8 >=0.6         :  1.6.2 (OK)
psutil >=0.3       :  3.3.0 (OK)
pyflakes >=0.5.0   :  1.0.0 (OK)
pygments >=1.6     :  2.0.2 (OK)
pylint >=0.25      :  None (NOK)
qtconsole >=4.0    :  4.1.1 (OK)
rope >=0.9.2       :  0.9.4 (OK)
sphinx >=0.6.6     :  1.3.1 (OK)
sympy >=0.7.3      :  0.7.6.1 (OK)
zmq >=2.1.11       :  15.2.0 (OK)

**Comment by blink1073** [comment]
@ccordoba12, I just upgraded to El Capitan.  I see the same symptoms, but #2932 fixes it.

**Comment by ccordoba12** [comment]
@blink1073, those are fantastic news!! Unfortunately I don't feel comfortable in backporting those changes to our stable branch and do a new release (2.3.9, that is).

What's your take on this? Should we backport or advise people to upgrade to 3.0beta3 (when its ready :-)?

**Comment by blink1073** [comment]
New OS requires new software I say :).

**Comment by nbara** [comment]
dear @ccordoba12, does 3.0beta2 already implement fix #2932? 

i use spyder for work, and 2.3.8 is completely unusable for me as is.

**Comment by ccordoba12** [comment]
Working on it, thanks for your patience :-) I'll try to merge that PR and release beta3 this week.

**Comment by parichayc** [comment]
Hi Everyone - first of all thank you very much for creating Spyder and keeping it open source. I am a Python newbie, and the first good tutorial I came across uses Spyder - and on OSX 10.11 I noticed the lag also.

Since I installed Spyder along with Anaconda (with Python 2.x) - is there an easy way for me to "upgrade" my Spyder to the beta3 once it is released? I am sorry if this is a naive question - please point me to the relevant documentation if I have jumped the gun here. I was hoping to only have my existing Spyder installation (which I launch form clicking on "Anaconda" and selecting the environment) and "applying" the beta to it, if possible.

Thank you.

**Comment by ccordoba12** [comment]
> Is there an easy way for me to "upgrade" my Spyder to the beta3 once it is released?

Yes, there is. I'll post the instructions here once beta3 is released :-)

**Comment by parichayc** [comment]
Awesome! Thank you very much indeed :)

**Comment by nknezek** [comment]
I'm having a lot of slowdown on my installation. latest version of spyder, pyqt 4.11, Mac OS X 10.11.3. Currently using python 2.7.

**Comment by zzhao212** [comment]
Changing Spyder-Preference-General-Qt windows style to be something other than Macintosh (which is the default) solved the issue for me.

**Comment by sho-87** [comment]
@zzhao212 None of the other styles seemed to fix the issue for me

**Comment by MetaDev** [comment]
@zzhao212 For me the "Plastique" style removed all lag while typing. Still some lag on the auto-complete (ctr-space) but overall much better. Thanks!

**Comment by kcarnold** [comment]
Unfortunately the merge of #2932 did _not_ fix the slowness for me :( spyder was usable for a while, perhaps longer than before though hard to say, but as I write this, bootstrapped right off the merge commit, text entry has become laggy again.

I just retested integrated and discrete GPU, internal and external display. Same problem. For an idea of the scale of the problem, here are two screen recordings:

https://www.dropbox.com/s/6hhxqmvhkn85u7u/laggy%20spyder%20discrete%20GPU.mov?dl=0
https://www.dropbox.com/s/fld8m8i6z6wdbx0/laggy%20spyder%20integrated%20GPU.mov?dl=0

**Comment by kcarnold** [comment]
As a follow-up: spyder's CPU usage hovers around 5-15% (!) when idle, spiking on text entry, and when I'm mashing on the keyboard as in the above videos, it gets up to 60%.

Also, switching themes to Plastique, as an earlier comment suggests, makes no difference.

**Comment by ccordoba12** [comment]
I think the problem is related not only to code completion but also to how we handle keyboard entries. I'll investigate further.

**Comment by talloniv** [comment]
Like @kcarnold and many others above, I'm using a newer retina-MBP and the Anaconda installation of Spyder (2.8.3) and am having extreme sluggishness in the editor to the point of making Spyder unusable.
The input lag is not observed in the IPython Console, but is prominent in both the editor and the standard Console (non-IPython, though to a lesser degree than the editor).

The sluggishness is observed immediately upon opening Spyder, though as someone said above, importing a library (such as import re) and using a method from a library (such as re.findall()) causes _extreme_ slowdown of text input in the editor.  This is reproducible 100% of the time, and takes effect isntantly.  This sluggishness remains permanently, rendering Spyder unusable.

Notes:
This happens even when only one file is open and it is a fresh python file.  
This happens even after a fresh reboot of the computer and no other programs are running.
This happens using both the integrated graphics card (Intel Iris Pro 1536 MB) and the discrete graphics card (AMD Radeon R9 M370X).
The memory usage reported by Spyder upon launch is usually 56% and hovers in the 50-60% range permanently, even when an empty 'new file' is all that is loaded.


**Info**:
2015 retina-Macbook Pro, 15"
Mac OS X 10.11.3 - El Capitan
2.8 GHz Intel Core i7
16 GB RAM

Spyder 2.3.8
Python 2.7.10 
Anaconda 2.3.0 (x86_64) (default, Oct 19 2015, 18:31:17) 

**Dependencies**:
IPython >=1.0      :  4.0.0 (OK)
jedi >=0.8.1;<0.9.0:  0.9.0 (NOK)
matplotlib >=1.0   :  1.5.0 (OK)
pandas >=0.13.1    :  0.17.1 (OK)
pep8 >=0.6         :  1.6.2 (OK)
psutil >=0.3       :  3.3.0 (OK)
pyflakes >=0.5.0   :  1.0.0 (OK)
pygments >=1.6     :  2.0.2 (OK)
pylint >=0.25      :  None (NOK)
qtconsole >=4.0    :  4.1.1 (OK)
rope >=0.9.2       :  0.9.4 (OK)
sphinx >=0.6.6     :  1.3.1 (OK)
sympy >=0.7.3      :  0.7.6 (OK)
zmq >=2.1.11       :  14.7.0 (OK)

Thanks for reopening this issue and continuing to attempt to resolve it.
Spyder is a wonderful IDE and I'd prefer to use it.

**Comment by blink1073** [comment]
Okay, I can reproduce this now, and it happens when either rope or jedi are used to make the first completion.  I'll look into it this weekend.

**Comment by ccordoba12** [comment]
@blink1073, questions:

1. Could it be a problem with PyQt4 that it's solved by moving to PyQt5?
2. @goanpeca mentioned we're checking for every keyboard event to filter certain keys. That could be causing the slowdown too.

**Comment by blink1073** [comment]
I verified that there is no slowdown for PyQt5:

```
conda uninstall pyqt 
conda uninstall qt
conda install -c spyder-ide pyqt5
```

**Comment by ccordoba12** [comment]
Ok, so for those of you who are using Anaconda, please follow @blink1073's commands, along with this last one:

    conda install -c spyder-ide spyder==3.0.0b2

and let us know if you experience the slowdown or not.

It's really important that we understand the source of the problem, and according to @blink1073 the problem is with current Qt version shipped with Anaconda, not with code completions, themes or anything else.

**Comment by ccordoba12** [comment]
I'm sorry, the series of commands is this:

    conda install -c spyder-ide spyder==3.0.0b2
    conda uninstall pyqt 
    conda uninstall qt
    conda install -c spyder-ide pyqt5

**Comment by corjos** [comment]
Worked for me (MBP mid-2010, El Capitan). Thanks for the support.

**Comment by psemyon** [comment]
Yep, these series worked for me also (MPB mid 2015, El Capitan). Thank you!

**Comment by ChrisAndre** [comment]
The PyQt5 with Spyder==3.0.0b2 Anaconda patch reduced the observed text buffer latency considerably right off the bat, on an El Capitan MBP. I do not notice any significant lag anymore, and it is not getting worse with time as far as I have tested it. My code introspection settings are all enabled. Much appreciated!

**Comment by ccordoba12** [comment]
Ok, those are fantastic news!! We'll be sure that our beta3 packages for Mac depend on pyqt5 then.

Thanks to all for your patience. We're still fixing some last bugs but we hope to release beta3 this weekend.

**Comment by talloniv** [comment]
@ccordoba12 I ran your modified series of commands and I can confirm that it completely resolved my lag issue (thus far).  I've not had enough time to do extensive work in Spyder with this change, yet, but so far there are no issues.

Thank you very much for working to fix the problem -- looking forward to future versions as you guys continue to improve this great IDE.
-JP

**Comment by nknezek** [comment]
I don't use Anaconda as I manage my software with homebrew and pip. I've upgraded to python3 with pyqt5, installed spyder 3.0.0b2 using 
`pip3 install --pre spyder`

The delay seems to be gone, which is great! There are other bugs in 3.0.0b2 that are annoying (unchangeable keyboard shortcut for hiding/showing console/editor that used to be used to switch between editor and console, an error in internal console when opening preferences, and others),  but I will submit them in individual bug reports. 

I guess it's time for me to make sure all my code runs in python3! Thanks for the feedback

**Comment by parichayc** [comment]
Hi, as I am new to Spyder and Python, I wanted to ask: if I do run the "official" commands as described by ccordoba12 on OSX (thank you very much for those), then at a later point in time, how should I keep my Anaconda (and Spyder, etc) updated? I mean in future can I just run "conda update anaconda" and expect it to work as it currently does, after I run the custom commands to remove the lag? Thank you for helping out.

**Comment by ccordoba12** [comment]
@nknezek, please report those bugs as soon as you can so I can fix them for beta3.

----

@parichayc, after running

    conda update anaconda

you will need to run only the first one, i.e.

    conda install -c spyder-ide spyder==3.0.0b3

for Spyder 3.0 beta3 and other betas until we release Spyder 3.0 final. This is because:

1. We are working to make `pyqt` and `pyqt5` work side by side in Anaconda, so you won't need to uninstall `qt` and `pyqt` to run Spyder with `pyqt5`.
2. Since beta3 Spyder is going to depend on the `pyqt5` package (instead of the `pyqt` one) for OS X to prevent this problem.

**Comment by danipoak** [comment]
I just wanted to chime in the with qt5 and running beta two spyder seems to be much much smoother and more pleasant to you. Thank you for helping us out @ccordoba12 ! I am running an old core2 duo macbook pro and it is even running well on that!

**Comment by AlJohri** [comment]
@ccordoba12 I'm having trouble with your instructions:

```
conda update anaconda
conda install -c spyder-ide spyder==3.0.0b2
conda uninstall pyqt # removed pyqt: 4.11.4-py27_1
conda uninstall qt # removed qt: 4.8.7-1
conda install -c spyder-ide pyqt5 # installs pyqt5: 5.5.0-py27_0 and pyqt5-5.5.0
```

I get this error when I run `spyder`:

```
Traceback (most recent call last):
  File "/Users/johria/.pyenv/versions/anaconda2-2.5.0/lib/python2.7/site-packages/spyderlib/spyder.py", line 3119, in main
    mainwindow = run_spyder(app, options, args)
  File "/Users/johria/.pyenv/versions/anaconda2-2.5.0/lib/python2.7/site-packages/spyderlib/spyder.py", line 3005, in run_spyder
    main.setup()
  File "/Users/johria/.pyenv/versions/anaconda2-2.5.0/lib/python2.7/site-packages/spyderlib/spyder.py", line 822, in setup
    message=_("Spyder Internal Console\n\n"
  File "/Users/johria/.pyenv/versions/anaconda2-2.5.0/lib/python2.7/site-packages/spyderlib/plugins/console.py", line 79, in __init__
    self.find_widget.set_editor(self.shell)
  File "/Users/johria/.pyenv/versions/anaconda2-2.5.0/lib/python2.7/site-packages/spyderlib/widgets/findreplace.py", line 250, in set_editor
    from spyderlib.qt.QtWebKit import QWebView
  File "/Users/johria/.pyenv/versions/anaconda2-2.5.0/lib/python2.7/site-packages/spyderlib/qt/QtWebKit.py", line 10, in <module>
    from PyQt5.QtWebKitWidgets import QWebPage, QWebView      # analysis:ignore
ImportError: No module named QtWebKitWidgets
```

**Comment by ccordoba12** [comment]
This string in your directories makes me suspicious: `.pyenv`. Are you mixing venvs and conda envs?

**Comment by AlJohri** [comment]
No conda envs. Just pyenv install latest anaconda 2.

Sent from Outlook Mobile




On Sat, Mar 19, 2016 at 9:55 AM -0700, "Carlos Cordoba" <notifications@github.com> wrote:












This string in your directories makes me suspicious: .pyenv. Are you mixing venvs and conda envs?

**Comment by ccordoba12** [comment]
Please use conda envs instead.

**Comment by MooMinIL** [comment]
New user here, I'm attempting to migrate from MATLAB on a MBP retina 15" mid-2015 with El Capitan and even after installing anaconda and updating to the Spyder beta i've been experiencing the lag.
Any ideas?

**Comment by talloniv** [comment]
I updated all components in the Anaconda launcher and it caused lag to reappear. So I simply re-ran the commands C. Cordova listed above in the terminal and it resolved all lag issues.

I'd recommend just running those again.

Best,
Jonpaul Wright

**Comment by ccordoba12** [comment]
@MooMinIL, it's not only necessary to update Spyder to it's latest beta, you need to install Qt5 and PyQt5 to fix the lagging issues.

To do this, you need to follow the comment @talloniv is referring to, which is: https://github.com/spyder-ide/spyder/issues/2724#issuecomment-190788309

**Comment by JiannanZhang** [comment]
I downloaded the anaconda 2 and then run these commands:
conda install -c spyder-ide spyder==3.0.0b2
conda uninstall pyqt 
conda uninstall qt
conda install -c spyder-ide pyqt5

After these commands, the navigator app and some files are gone so I cannot open the navigator app or Spyder anymore. How should I fix this?

**Comment by ccordoba12** [comment]
Navigator and Spyder should work after this change, but if you want to go unroll these changes, you need to do

    conda remove qt5 pyqt5
    conda install qt pyqt

**Comment by JiannanZhang** [comment]
@ccordoba12 Thanks a lot for your reply! 
But if I undo these commands, it is the same as I just install spyder 3.0 beta, which has the cursor-lag problem. You mean other people can fix the issue just by those four commands?

**Comment by sho-87** [comment]
any idea on when these updates will be rolled into the next official release instead of a beta version? Or is this something that needs to happen on the Anaconda side of things?

**Comment by MarKo9** [comment]
Following the steps as suggested

```
conda install -c spyder-ide spyder==3.0.0b2
conda uninstall pyqt 
conda uninstall qt
conda install -c spyder-ide pyqt5
```

the navigator shortcut/icon seems to dissapear after the  "conda uninstall pyqt" step. Anyone else having this issue?

**Comment by ccordoba12** [comment]
We're working to release this fix in beta3, but it caught us in the midst of a new Qt5 version (Qt 5.6) for which we need to update Spyder first.

Sorry guys for the big delay in addressing this one :-(

**Comment by ccordoba12** [comment]
> the navigator shortcut/icon seems to dissapear after the "conda uninstall pyqt" step. Anyone else having this issue?

The problem is Navigator is currently based in Qt/PyQt (instead of Qt5/PyQt5 :-). But for Spyder beta3 there won't be any need to run

    conda uninstall pyqt 
    conda uninstall qt

which will Navigator intact ;-)

**Comment by AidanRoss** [comment]
`conda install -c spyder-ide spyder==3.0.0b2`
`conda uninstall pyqt `
`conda uninstall qt`
`conda install -c spyder-ide pyqt5`

Tried the following and this did not fix my problem. 3.0.0b2 seemed to be even slower. Any news on further updates? @ccordoba12

**Comment by gamehere007** [comment]
I think once we install pyqt5 on Anaconda, the navigator cannot be opened. How do we fix this issue?

**Comment by LRV72** [comment]
When I uninstall pyqt, python3.5.1 is downgraded to python2.7.11. Is it possible to avoid this?

**Comment by zzhao212** [comment]
@ccordoba12 Regarding the solution you proposed, on my machine `conda uninstall pyqt` reverts spyder 3.0.0b2 back to 2.0.0. What's worse, `conda uninstall qt` actually removes spyder altogether. Any idea why this might be? Thanks!

**Comment by ccordoba12** [comment]
This week we (at Continuum) updated our `qt` and `pyqt` packages, so now there's no need to run `conda uninstall pyqt` to have this working :-)

**Comment by ccordoba12** [comment]
So you need to run

    conda update qt
    conda update pyqt

to get those updates.

**Comment by zzhao212** [comment]
@ccordoba12 Are you saying `conda install -c spyder-ide spyder==3.0.0b2` alone should solve the problem?

**Comment by ccordoba12** [comment]
And the commands I posted in my previous comment

**Comment by zzhao212** [comment]
@ccordoba12 Sorry just saw your second comment. Should I install spyder 3.0.0b2 or 2.0.0 before updating qt and pyqt?

**Comment by ccordoba12** [comment]
The order is this

    conda update qt
    conda update pyqt
    conda install -c spyder-ide pyqt5
    conda install -c spyder-ide spyder==3.0.0b2

**Comment by zzhao212** [comment]
@ccordoba12 Unfortunately this didn't resolve the issue for me. Below are the steps I took starting from scratch.

1. `conda create --name test numpy pandas`: Create new test env
2. `conda install -c spyder-ide spyder==3.0.0b2`: Install newest spyder beta
3. `conda update qt` and `conda update pyqt`: Make sure the two are up-to-date (which they are)
4. `conda install -c spyder-ide pyqt5`

Choppiness still didn't go away. One thing I noticed was that even after I installed pyqt5, the bottom of "python - About python" still showed Qt4 and PyQt4. I'm fairly new to this so please tell me anything else I should provide that could help you pinpoint the problem. Thanks!

**Comment by ccordoba12** [comment]
@zzhao212, if you're new to this, please wait until Anaconda 4.1 is released at the end of this month, then jus reinstall it. That version will come with Spyder 3.0 and Qt5.

If you don't want to wait, please follow my instructions in the *exact* same order I gave them, and drop your step 1.

**Comment by ccordoba12** [comment]
@zzhao212, I'm very sorry, I just noticed my instructions are wrong! :-) I'll post new instructions tomorrow after releasing Spyder 3.0 beta 3.

**Comment by zzhao212** [comment]
@ccordoba12 Thanks! Is it going to be on this thread?

**Comment by ccordoba12** [comment]
Yep, and I'll also post them in our release announcement (in our mailing list :-)

**Comment by gamehere007** [comment]
@ccordoba12  

hi! ccordoba12. Can you add me to your mailing list?

tks!

**Comment by ccordoba12** [comment]
This is the address of our mailing list

https://groups.google.com/forum/#!forum/spyderlib

You can subscribe yourselves :-)

**Comment by ccordoba12** [comment]
Ok, I released Spyder 3.0 beta3, so the new instructions are:

* Run these commands in your root environment
* `conda update qt pyqt`
* `conda install -c spyder-ide spyder==3.0.0b3`

The second command should install the `qt5` and `pyqt5` packages (but only for OS X, not for other operating systems). Those are the ones needed to fix this problem.

**Comment by gamehere007** [comment]
@ccordoba12 
hi! I got an error from my terminal:

Error:  Package missing in current osx-64 channels: 
  - spyder ==3.0.0b3

So what is happening

**Comment by ccordoba12** [comment]
Sorry, I uploaded the packages to the wrong channel.

Please try again, now things should be working fine :-)

**Comment by zzhao212** [comment]
@ccordoba12 Problem solved! Thanks so much for your continued help. Please keep up the good work.

**Comment by gamehere007** [comment]
@ccordoba12 

Hi! I followed the instruction. But it is still the same problem that I can not open Navigator once qt5 and pyqt 5 have been installed.  

Thanks!

**Comment by AidanRoss** [comment]
The same issue occurs for me when I try to open Navigator, every time I get an error that says "Navigator Quit unexpectedly"

**Comment by ccordoba12** [comment]
This is a problem with Navigator not working correctly with Qt5 (although it shouldn't have any problems).

@goanpeca, please start testing Navigator with Qt5 in Mac and try to fix this error :-)

**Comment by zzhao212** [comment]
@ccordoba12 Now that the sluggishness is gone, there seems to be a new issue. If I open up spider, leave it hanging and do nothing at all, I get the spinning wheel of death after roughly 10min and need to force quit. Any idea which areas I need to look into to provide a more accurate description of the problem? Thanks.

**Comment by Suinga** [comment]
@zzhao212 I have exactly same problem. Any idea to fix this, please? Many thanks.

**Comment by ccordoba12** [comment]
Guys, please open a new issue about this problem. I consider the 
sluggishness problem in the Editor to be (finally) solved.

**Comment by zzhao212** [comment]
@ccordoba12 It seems that Anaconda4.1 doesn't ship with spyder3 and qt5. Is there plan to reflect those upgrades in a later release of Anaconda? Thanks.

**Comment by ycemsubakan** [comment]
Hi, I have followed the steps suggested above (involving uninstalling qt and qtpy) , now I am getting this error: qtpy.PythonQtError: No Qt bindings could be found  - I have installed qt and qtpy, but I am still getting the same error. Thanks!
